rotkehlchen/db/filtering.py

```diff
@@ -98,23 +98,25 @@ class DBETHTransactionJoinsFilter(DBFilter):
     should_join_events: bool = False
 
     def prepare(self) -> Tuple[List[str], List[Any]]:
-        query_filter, bindings = '', []
+        query_filters, bindings = [], []
         if self.should_join_events is True:
-            query_filter += (
-                ' INNER JOIN '
-                '(SELECT event_identifier, counterparty, asset FROM history_events) '
-                'ON hex(ethereum_transactions.tx_hash)=substr(event_identifier, 3)'
+            query_filters.append(
+                'LEFT JOIN (SELECT event_identifier, counterparty, asset FROM history_events) '
+                'ON hex(ethereum_transactions.tx_hash)=substr(event_identifier, 3)',
             )
         if self.addresses is not None:
             questionmarks = '?' * len(self.addresses)
-            query_filter += (
-                f' INNER JOIN ethtx_address_mappings ON '
+            query_filters.append(
+                f'INNER JOIN ethtx_address_mappings WHERE '
                 f'ethereum_transactions.tx_hash=ethtx_address_mappings.tx_hash AND '
-                f'ethtx_address_mappings.address IN ({",".join(questionmarks)})'
+                f'ethtx_address_mappings.address IN ({",".join(questionmarks)})',
             )
             bindings += self.addresses
+        else:
+            query_filters.append('WHERE 1')
```

Comment:
We need this line here because filters expect the join clause to end with a WHERE clause.

Reply:
Hmm kind of ugly but let's see if this works out. Your comment though should 100% be in the code and not just Github as the reader (which can be you in 3 months) may read this and think: "WTF?" :thinking:

Reply:
Oh yes, I also thought about adding this as a code comment when was writing it in  GitHub :)